querydsl-core/src/main/java/com/querydsl/core/types/dsl/PathBuilder.java

```diff
@@ -44,9 +45,9 @@
 
     private static final long serialVersionUID = -1666357914232685088L;
 
-    private final Map<String, PathBuilder<?>> properties = Maps.newHashMap();
+    private final ConcurrentHashMap<SimpleEntry<String, Class<?>>, PathBuilder<?>> properties = new ConcurrentHashMap<SimpleEntry<String, Class<?>>, PathBuilder<?>>();
```

Comment:
The highest we can go is `ConcurrentMap`, otherwise it will compile against `Map.putIfAbsent`, which is only Java 8.
But @robertandrewbain said that elsewhere we also have the `ConcurrentHashMap`.